ksql-execution/src/main/java/io/confluent/ksql/services/KafkaTopicClient.java

```diff
@@ -61,8 +61,7 @@ default void createTopic(
    * {@code replicationFactor}
    *
    * @param topic             name of the topic to create
-   * @param replicationFactor the replication factor for the new topic, or
-   *                          {@link io.confluent.ksql.topic.TopicProperties#DEFAULT_REPLICAS}
+   * @param replicationFactor the replication factor for the new topic, or (TODO(rohan): fixme)
```

Comment:
Did you mean to move `TopicProperties` to `execution` as well? In general, I'm a little confused now about the difference between the engine and the execution modules. I feel like anything contacting external services (e.g. ConnectClient/KafkaTopicClient) belong in the engine, no?

Reply:
> Did you mean to move TopicProperties to execution as well? In general, I'm a little confused now about the difference between the engine and the execution modules. I feel like anything contacting external services (e.g. ConnectClient/KafkaTopicClient) belong in the engine, no?

💯 

FWIW what's here now is just the interface, not the actual service implementations themselves. The main reason I had to do this was because we need a schema registry client supplier to build serdes. But this requires splitting out KsqlQueryBuilder into something the execution plan can use vs something that holds all the context we need in the engine when building the plan. This particular PR was getting really big so I kind of drew the line there and moved the whole KsqlQueryBuilder into ksql-execution for now, with the intent to split it up in a follow-up. Ideas welcome on how we can do this better.